elytron/src/main/java/org/wildfly/extension/elytron/PolicyDefinitions.java

```diff
@@ -287,292 +276,131 @@ protected void recordCapabilitiesAndRequirements(OperationContext context, Attri
-
-            // add the 'Roles' group to the subject containing the identity's mapped roles.
-            Group rolesGroup = new SimpleGroup("Roles");
-            for (String role : securityIdentity.getRoles()) {
-                rolesGroup.addMember(new NamePrincipal(role));
-            }
-            subject.getPrincipals().add(rolesGroup);
-
-            // add a 'CallerPrincipal' group containing the identity's principal.
-            Group callerPrincipalGroup = new SimpleGroup("CallerPrincipal");
-            callerPrincipalGroup.addMember(securityIdentity.getPrincipal());
-            subject.getPrincipals().add(callerPrincipalGroup);
-
-            // process the identity's public and private credentials.
-            for (Credential credential : securityIdentity.getPublicCredentials()) {
-                if (credential instanceof PublicKeyCredential) {
-                    subject.getPublicCredentials().add(credential.castAs(PublicKeyCredential.class).getPublicKey());
-                }
-                else if (credential instanceof X509CertificateChainPublicCredential) {
-                    subject.getPublicCredentials().add(credential.castAs(X509CertificateChainPublicCredential.class).getCertificateChain());
-                }
-                else {
-                    subject.getPublicCredentials().add(credential);
-                }
-            }
+    private static PolicyConfigurationFactory newPolicyConfigurationFactory(String className, ClassLoader classLoader) throws PolicyContextException, ClassNotFoundException {
+        final ClassLoader original = Thread.currentThread().getContextClassLoader();
 
-            for (Credential credential : doPrivileged((PrivilegedAction<IdentityCredentials>) securityIdentity::getPrivateCredentials)) {
-                if (credential instanceof PasswordCredential) {
-                    addPrivateCredential(subject, credential.castAs(PasswordCredential.class).getPassword());
-                }
-                else if (credential instanceof SecretKeyCredential) {
-                    addPrivateCredential(subject, credential.castAs(SecretKeyCredential.class).getSecretKey());
-                }
-                else if (credential instanceof KeyPairCredential) {
-                    addPrivateCredential(subject, credential.castAs(KeyPairCredential.class).getKeyPair());
-                }
-                else if (credential instanceof X509CertificateChainPrivateCredential) {
-                    addPrivateCredential(subject, credential.castAs(X509CertificateChainPrivateCredential.class).getCertificateChain());
-                }
-                else {
-                    addPrivateCredential(subject, credential);
-                }
+        try {
+            Thread.currentThread().setContextClassLoader(classLoader);
+            System.setProperty(PolicyConfigurationFactory.class.getName() + ".provider", className);
```

Comment:
should this sys prop remain? or should it be discarded at some point?

Reply:
TBH after setting it and triggering the initialisation clearing it has no effect, without restarting the process it is not possible to clear the previous result.

TBH compared to some of the other specs this seems to be a gap in JACC that was can't manually swap in a different implementation / instance.

Reply:
ok, thanks for the explanation